src/controllers/sign.out.controller.ts

```diff
@@ -29,17 +28,34 @@ export const post = (req: Request, res: Response, next: NextFunction) => {
     logger.debugRequest(req, `POST ${config.SIGN_OUT_PAGE}`);
     const previousPage = req.body["previousPage"];
 
-    if (!previousPage.includes(config.REGISTER_AN_OVERSEAS_ENTITY_URL)){
+    if (!previousPage.startsWith(config.REGISTER_AN_OVERSEAS_ENTITY_URL)){
       throw createAndLogErrorRequest(req, `${previousPage} page is not part of the journey!`);
     }
 
     if (req.body["sign_out"] === 'yes') {
       return res.redirect(config.ACCOUNTS_SIGN_OUT_URL);
     }
 
-    return res.redirect(previousPage);
+    return safeRedirect(res, previousPage);
   } catch (error) {
     logger.errorRequest(req, error);
     next(error);
   }
 };
+
+function getPreviousPageUrl(req: Request) {
+  const headers = req.rawHeaders;
+  const absolutePreviousPageUrl = headers.filter(item => item.includes(config.REGISTER_AN_OVERSEAS_ENTITY_URL))[0];
+
+  // Don't attempt to determine a relative previous page URL if no absolute URL is found
+  if (!absolutePreviousPageUrl) {
+    return absolutePreviousPageUrl;
+  }
+
+  const startingIndexOfRelativePath = absolutePreviousPageUrl.indexOf(config.REGISTER_AN_OVERSEAS_ENTITY_URL);
```

Comment:
Would it be safer to use:
const relativePreviousPageUrl = new URL(absolutePreviousPageUrl).pathname;

Reply:
Not sure about safer, but definitely simpler. However, if the `absolutePreviousPageUrl` contains a sub-heading reference at the end, this isn't returned by the `pathname` function.

E.g. If `absolutePreviousPageUrl` is **http://hostname/register-an-overseas-entity/new-page#sub-heading** then only **/register-an-overseas-entity/new-page** is returned.

What do we think (@AndreKramer7654 @markpit ) is this ok?

Reply:
ok

Reply:
👍